packages/flutter/lib/src/widgets/editable_text.dart

```diff
@@ -2023,12 +2023,28 @@ class EditableTextState extends State<EditableText> with AutomaticKeepAliveClien
     // as the pre-formatting value of the previous pass (repeat call).
     final bool textChanged = _value?.text != value?.text;
     final bool isRepeat = value == _lastFormattedUnmodifiedTextEditingValue;
+    final bool isRepeatText = value?.text == _lastFormattedUnmodifiedTextEditingValue?.text;
 
-    if (textChanged && widget.inputFormatters != null && widget.inputFormatters.isNotEmpty) {
+    // There's no need to format when the value start to composing and repeat composing.
+    final bool isValueComposing = value != null && value.composing != const TextRange(start: -1, end: -1);
+    final bool isLastUnmodifiedValueComposing = _lastFormattedUnmodifiedTextEditingValue != null &&
+      _lastFormattedUnmodifiedTextEditingValue.composing != const TextRange(start: -1, end: -1);
+    final bool isRepeatComposing = isValueComposing == isLastUnmodifiedValueComposing;
+
+    if (((!isRepeatText && textChanged) || (!isRepeatComposing && !isValueComposing)) &&
+        widget.inputFormatters != null &&
+        widget.inputFormatters.isNotEmpty) {
       // Only format when the text has changed and there are available formatters.
       // Pass through the formatter regardless of repeat status if the input value is
       // different than the stored value.
       for (final TextInputFormatter formatter in widget.inputFormatters) {
+        // Skip format update while the editing value is composing with fixed length.
+        if ((formatter is LengthLimitingTextInputFormatter) &&
```

Comment:
If the problem is with that specific formatter, can we apply the fix to the formatter instead?

Reply:
Done it, the fix is now land of the specific formatter.